sklearn/utils/estimator_checks.py

```diff
@@ -927,11 +932,19 @@ def check_sample_weights_invariance(name, estimator_orig, kind="ones"):
 @ignore_warnings(category=(FutureWarning, UserWarning))
 def check_dtype_object(name, estimator_orig):
     # check that estimators treat dtype object as numeric if possible
-    rng = np.random.RandomState(0)
-    X = _pairwise_estimator_convert_X(rng.rand(40, 10), estimator_orig)
+    seed = 0
+    if name == 'QuadraticDiscriminantAnalysis':
+        X, y = make_classification(random_state=seed, n_samples=40,
+                                   n_informative=8, n_features=10,
+                                   n_classes=4)
```

Comment:
We try not to special case estimators here. Is there no `reg_param` we can use to make this work?

Reply:
sadly no. With random data, the check is always failing no matter how large the reg_param, I tried quite a few different ones. That's why I also mentioned in the earlier param that changing the error to a warning might be a better solution.

Reply:
I changed the error to a warning with the latest commit and reverted back to the original state with estimator checks.

Reply:
It seems like `LinAlgWarning` was introduced at SciPy==1.1.0 and the CI pipelines for Python 3.6 are using SciPy==0.19. SciPy actually support Python 3.6 until version 1.6.0. Is it possible to upgrade the CI requirements?